p2p/switch.go

```diff
@@ -320,6 +320,9 @@ func (sw *Switch) Peers() IPeerSet {
 // TODO: make record depending on reason.
 func (sw *Switch) StopPeerForError(peer Peer, reason interface{}) {
 	sw.Logger.Error("Stopping peer for error", "peer", peer, "err", reason)
+	if peer == nil {
```

Comment:
how's this possible?

Reply:
In the test, it happened because it fetches the peer from the switch, but the peer is not registered in the switch yet, so it returns a `nil` peer which is then stopped for error.

Reply:
should we add a comment that this condition is there because of the test?

Reply:
It's not strictly because of the test, it's just good to guard against `nil` panics, especially in concurrent code which can be a bit hard to reason about.

Reply:
such guards a) litter the code b) hide the errors from the developers. I'd advocate against using them